src/vm/database/clarity_db.rs

```diff
@@ -39,32 +41,89 @@ pub enum StoreType {
 }
 
 pub struct ClarityDatabase<'a> {
-    pub store: RollbackWrapper<'a>
+    pub store: RollbackWrapper<'a>,
+    headers_db: &'a dyn HeadersDB,
+}
+
+pub trait HeadersDB {
+    fn get_stacks_block_header_hash_for_block(&self, id_bhh: &BlockHeaderHash) -> Option<BlockHeaderHash>;
+    fn get_burn_header_hash_for_block(&self, id_bhh: &BlockHeaderHash) -> Option<BurnchainHeaderHash>;
+    fn get_vrf_seed_for_block(&self, id_bhh: &BlockHeaderHash) -> Option<VRFSeed>;
+    fn get_burn_block_time_for_block(&self, id_bhh: &BlockHeaderHash) -> Option<u64>;
+}
+
+fn get_stacks_header_info(conn: &DBConn, id_bhh: &BlockHeaderHash) -> Option<StacksHeaderInfo> {
+    conn.query_row("SELECT * FROM block_headers WHERE index_block_hash = ?",
+                   [id_bhh].iter(),
+                   |x| StacksHeaderInfo::from_row(x).expect("Bad stacks header info in database"))
+        .optional()
+        .expect("Unexpected SQL failure querying block header table")
+}
+
+impl HeadersDB for DBConn {
+    fn get_stacks_block_header_hash_for_block(&self, id_bhh: &BlockHeaderHash) -> Option<BlockHeaderHash> {
+        get_stacks_header_info(self, id_bhh)
+            .map(|x| x.anchored_header.block_hash())
+    }
+    
+    fn get_burn_header_hash_for_block(&self, id_bhh: &BlockHeaderHash) -> Option<BurnchainHeaderHash> {
+        get_stacks_header_info(self, id_bhh)
+            .map(|x| x.burn_header_hash)
+    }
+
+    fn get_burn_block_time_for_block(&self, _id_bhh: &BlockHeaderHash) -> Option<u64> {
+        panic!("Block time data not available in burn header db")
```

Comment:
Adding this is going to take more than adding a column for it in the headers db.  In particular, the code in `src/burnchains` will need to be updated so that the timestamp is something the burnchain indexer picks up from burnchain block headers.  We might want to just leave that for a separate PR.

Reply:
Alternatively, I can contribute to this PR and add it.  Open to either approach.

Reply:
I'll just leave it as a panicking stub function for now -- I'll add a stub for the miner address as well (#1131)